gradle-plugin/plugin/src/main/java/com/yelp/codegen/plugin/GenerateTask.kt

```diff
@@ -20,112 +23,111 @@ const val DEFAULT_NAME = "defaultname"
 const val DEFAULT_PACKAGE = "com.codegen.default"
 const val DEFAULT_OUTPUT_DIR = "/gen"
 
-open class GenerateTask : DefaultTask() {
+abstract class GenerateTask : DefaultTask() {
 
     init {
         description = "Run the Swagger Code Generation tool"
         group = BasePlugin.BUILD_GROUP
     }
 
-    @Input
-    @Optional
-    @Option(option = "platform", description = "Configures the platform that is used for generating the code.")
-    var platform: String? = null
-
-    @Input
-    @Optional
-    @Option(option = "packageName", description = "Configures the package name of the resulting code.")
-    var packageName: String? = null
-
-    @Input
-    @Optional
-    @Option(option = "specName", description = "Configures the name of the service for the Swagger Spec.")
-    var specName: String? = null
-
-    @Input
-    @Optional
-    @Option(option = "specVersion", description = "Configures the version of the Swagger Spec.")
-    var specVersion: String? = null
-
-    @InputFile
-    @Option(option = "inputFile", description = "Configures path of the Swagger Spec.")
-    lateinit var inputFile: File
-
-    @OutputDirectory
-    @Optional
-    @Option(option = "outputDir", description = "Configures path of the Generated code directory.")
-    var outputDir: File? = null
-
-    @InputFiles
-    @Optional
-    @Option(option = "extraFiles",
+    @get:Input
+    @get:Optional
+    @get:Option(option = "platform", description = "Configures the platform that is used for generating the code.")
```

Comment:
another way to put these could be:
```kotlin
@get:[Input Option(...)]
```
https://kotlinlang.org/docs/reference/annotations.html#annotation-use-site-targets
more of a style choice than anything, just throwing it out there.

Reply:
I'm new to this `Option` annotation actually. Is that a common thing to do ? Define task inputs from the command line ?

Reply:
> I'm new to this Option annotation actually

In our case we wanted to have this setup inside a docker container. So having a command line API made easier to configure the task from an external environment.